python_ta/contracts/__init__.py

```diff
@@ -1,40 +1,70 @@
 from typing import Any, List, Optional
 import typing
+import inspect
 import wrapt
 
 
-class InvalidPrecondition(Exception):
-    """Exception raised when precondition is invalid
+class InvalidCondition(Exception):
```

Comment:
It's a good idea to make this name more general now; let's call this `InvalidAssertion` instead, as an indicator that the strings are fundamentally wrapping `assert`s.

Reply:
Similarly below, when you've used "condition" let's change the generic name to "assertion", e.g. `parse_assertions` as the function name.

Reply:
Done